test/tap/lifecycle-path.js

```diff
@@ -32,46 +32,88 @@ test('setup', function (t) {
 })
 
 test('make sure the path is correct, without directory of current node', function (t) {
-  checkPath(false, false, t)
+  checkPath({
+    withDirOfCurrentNode: false,
+    prependNodePathSetting: false
+  }, t)
 })
 
 test('make sure the path is correct, with directory of current node', function (t) {
-  checkPath(true, false, t)
+  checkPath({
+    withDirOfCurrentNode: true,
+    prependNodePathSetting: false
+  }, t)
 })
 
 test('make sure the path is correct, with directory of current node but ignored node path', function (t) {
-  checkPath(true, true, t)
+  checkPath({
+    withDirOfCurrentNode: true,
+    prependNodePathSetting: true
+  }, t)
 })
 
 test('make sure the path is correct, without directory of current node and automatic detection', function (t) {
-  checkPath(false, 'auto', t)
+  checkPath({
+    withDirOfCurrentNode: false,
+    prependNodePathSetting: 'auto'
+  }, t)
 })
 
 test('make sure the path is correct, with directory of current node and automatic detection', function (t) {
-  checkPath(true, 'auto', t)
+  checkPath({
+    withDirOfCurrentNode: true,
+    prependNodePathSetting: 'auto'
+  }, t)
 })
 
 test('make sure the path is correct, without directory of current node and warn-only detection', function (t) {
-  checkPath(false, 'warn-only', t)
+  checkPath({
+    withDirOfCurrentNode: false,
+    prependNodePathSetting: 'warn-only'
+  }, t)
 })
 
 test('make sure the path is correct, with directory of current node and warn-only detection', function (t) {
-  checkPath(true, 'warn-only', t)
+  checkPath({
+    withDirOfCurrentNode: true,
+    prependNodePathSetting: 'warn-only'
+  }, t)
+})
+
+test('make sure there is no warning with a symlinked node and warn-only detection', function (t) {
+  checkPath({
+    withDirOfCurrentNode: 'extra-node',
```

Comment:
I originally made this a `boolean+` for the sake of not adding more arguments, but since you're switching to a config object, do you mind making `withDirOfCurrentNode` a plain bool, and adding an extra config for the `extra-node` bit?